src/gluonts/model/rotbaum/_predictor.py

```diff
@@ -0,0 +1,175 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+# Standard library imports
+from enum import Enum
+from typing import Iterator, List, Optional, Tuple, Union
+
+# Third-party imports
+import numpy as np
+import pandas as pd
+from itertools import chain
+
+# First-party imports
+from gluonts.core.component import validated
+from gluonts.dataset.common import Dataset
+from gluonts.model.forecast import Forecast, SampleForecast
+from gluonts.model.predictor import RepresentablePredictor
+from gluonts.support.pandas import forecast_start
+
+# Relative imports
+from ._preprocess import PreprocessOnlyLagFeatures
+from ._model import QRX
+
+
+class RotbaumForecast(Forecast):
+    """
+    Implements the quantile function in Forecast for TreePredictor,
+    as well as a new estimate_dists function for estimating a sampling of the
+    conditional distribution of the value of each of the steps in the
+    forecast horizon (independently).
+    """
+
+    @validated()
+    def __init__(
+        self,
+        models: List,
+        featurized_data: List,
+        start_date: pd.Timestamp,
+        freq,
+        prediction_length: int,
+    ):
+        self.models = models
+        self.featurized_data = featurized_data
+        self.start_date = start_date
+        self.freq = freq
+        self.prediction_length = prediction_length
+        self.item_id = None
+
+    @validated()
+    def quantile(self, q: float) -> np.array:
+        """
+        Returns np.array, where the i^th entry is the estimate of the q
+        quantile of the conditional distribution of the value of the i^th
+        step in the forecast horizon.
+        """
+        assert 0 <= q <= 1
+        return np.array(
+            list(
+                chain(
```

Comment:
There is [`chain.from_iterable()`](https://docs.python.org/3/library/itertools.html#itertools.chain.from_iterable).

I.e. you should be able to do

```python
np.array(list(chain.from_iterable(model.predict(self.featurized_data, q) for model in self.model))
```

Reply:
Done.